plugin.video.mlbtv/resources/lib/mlbmonitor.py

```diff
@@ -0,0 +1,259 @@
+                    if start_inning == 0:
+                        try:
+                            total_stream_time = player.getTotalTime()
+                            xbmc.log('Skip monitor for ' + content_id + ' total stream time ' + str(timedelta(seconds=total_stream_time)))
+                        except:
+                            pass
+                current_time = player.getTime()
+                # make sure we're not paused, and current time is valid (less than 10 hours) -- sometimes Kodi was returning a crazy large current time as the stream was starting
+                if current_time > 0 and current_time != last_time and current_time < 36000:
+                    last_time = current_time
+                    # remove any past skip markers so user can seek backward freely
+                    while len(skip_markers) > 0 and current_time > skip_markers[0][1]:
+                        xbmc.log("Skip monitor for " + content_id + " removed skip marker at " + str(skip_markers[0][1]) + ", before current time " + str(current_time))
+                        skip_markers.pop(0)
+                    # seek to end of break if we fall within skip marker range, then remove marker so user can seek backward freely
+                    if len(skip_markers) > 0 and current_time >= skip_markers[0][0] and current_time < skip_markers[0][1]:
+                        xbmc.log("Skip monitor for " + content_id + " processed skip marker at " + str(skip_markers[0][1]))
+                        player.seekTime(skip_markers[0][1])
+                        skip_markers.pop(0)
+                        # since we just processed a skip marker, we can delay further processing a little bit
+                        xbmc.sleep(2000)
+                    # if we've run out of skip markers and it's a live event and we're skipping things, check for more
+                    if len(skip_markers) == 0 and is_live == True and skip_type > 0:
+                        # refresh current time, and look ahead slightly
+                        current_time = player.getTime() + 10
+                        xbmc.log('Skip monitor for ' + content_id + ' refreshing skip markers from ' + str(current_time))
+                        skip_markers = self.get_skip_markers(skip_type, content_id, game_pk, current_time)
+                        xbmc.log('Skip monitor for ' + content_id + ' refreshed skip markers : ' + str(skip_markers))
+            else:
+                if self.stream_started == False:
+                    xbmc.log("Skip monitor for " + content_id + " waiting for stream to start")
+                    self.stream_start_tries -= 1
+                    if self.stream_start_tries < 1:
+                        xbmc.log("Skip monitor for " + content_id + " closing due to stream not starting")
+                        break
+
+            # wait 1 second until next loop iteration
+            xbmc.sleep(1000)
```

Comment:
Needed? You already wait at the beginning of the method. Maybe you want to waitForAbort here and just check abortRequested in the beginning?

Reply:
@tonywagner Can you verify this is needed?

Reply:
I think you are right, that's probably not necessary. I will test and likely drop it in the next release.